### ISSUE TITLE
Mutation of Input Coordinates

### ISSUE THREAD

**Issue by foxxyz** [proj4js/proj4js#474]
First off - thanks for a great library! I love it and use it a lot.

After noticing a strange bug in my application the other day, I traced it back to `proj4` and noticed that under some circumstances, the input coordinates are mutated, which I totally did not expect.

In my case, I'm passing in a 3-element array, such as `[x, y, z]` and when passed to `proj4` it actually removes the third element. Here's a reproduction:

```
import proj4 from 'proj4'
const projector = proj4('EPSG:3857')
const b = [0, 0, 2000]
projector.forward(b)
console.log(b)
// will print out [0, 0]
```

I believe this is happening on [line 17 in `core.js`](https://github.com/proj4js/proj4js/blob/2179190c3d2ad724dcd23f045933dd7de54753e2/lib/core.js#L17) at the `splice(..)` call.

Is this expected behavior? If not - would you accept a PR to remedy this?

**Comment by ahocevar** [comment]
Thanks for reporting this bug, @foxxyz. Will be fixed with #475.

**Comment by foxxyz** [comment]
Wow! Thank you for the super quick response and fix! 🎉 

Appreciate you 🙏🏽
